prime-router/src/main/kotlin/Translator.kt

```diff
@@ -61,6 +61,9 @@ class Translator(private val metadata: Metadata) {
         }
         val filteredReport = input.filter(filterAndArgs)
 
+        // Always succeed in translating an empty report after filtering (even if the mapping process would fail)
+        if (filteredReport.isEmpty()) return buildEmptyReport(receiver, input)
+
```

Comment:
If after filtering, there is an empty report, this can always be mapped to another empty report schema. Simplifies the code further down the pipeline.